src/engine/Math/vector-view.ts

```diff
@@ -0,0 +1,39 @@
+import { Vector } from '../Algebra';
+
+
+export interface VectorViewOptions<T> {
+  data: T;
+  getX: (source: T) => number;
+  getY: (source: T) => number;
+  setX: (source: T, x: number) => void;
+  setY: (source: T, y: number) => void;
+}
+export class VectorView<T> extends Vector {
```

Comment:
I don't think I understand the purpose of having to pass a data source. When you create a VectorView, implementing the getters and setters will let you close over any variables you'd need to access anyway.

If this is to help with closures, have you measured the impact? Otherwise I'd suggest keeping it simpler and not having to pass a data source (and therefore, don't need first arg of any of these methods).

Reply:
Ah true, I could definitely close over the source. I'll give that a shot :+1: